### DISCUSSION TITLE
triggering PopupInput from Layout View on Web Remote with correct display_handle

### DISCUSSION THREAD

**Discussion by lukas-runge** [hossimo/GMA3Plugins#23] [Q&A] [answered]
Hey everyone,

thank you for helping me out on my last issue that I was able to fix with your help! @hossimo 
The title pretty much says everything already but here is a more in depth explaination on what I think happens:

- If I trigger `PopupInput` oder any Input for that matter from the command line or the plugin pool the execution context / `display_handle` is set properly. The Popup opens on the Web Remote.
- I can't call the Plugin when it is assigned to a Layout View, because the call action is missing in the action assignment dropdown of any item assigned to Layout View.
- Calling the Plugin from a Macro works (`Call Plugin 1`) but it does not respect the context in which the button (now assigned to the beforementioned Layout View) has been pressed in. Thus the Popup opens on the onPC rather than on the Web Remote display.

Is there any way to workaround this? Maybe reading the webremotes display_handle from somewhere else in the system so the popup always opens on the Web Remote no matter what? Could that or something else maybe possible and what does it take to archive it?

**Thank you guys (especially @hossimo) again for the great help and fast answers on here - this is awesome!**
Kind regards
@lukas-runge

**Comment by hossimo** [comment]
Interesting, To be honest I'm not sure why this happens or if it is perhaps a bug, perhaps this is something that should be brought up on the official forums.

However as an example, in 1.4 if I call `Echo( HandleToInt( display_handle ) )` i get (the actual numbers are meaningless but their relationship is maningfull):

- Plugin Pool
  - oPC - 838861405
  - WebRemote  - 26608665349
- Macro
  - oPC - 838861405
  - WebRemote - 838861405


Also if I run this code, it kind of confirms that Macros on the web remote behave differently:
``` lua
  local d = GetDisplayCollect()
    for i = 1, d.count, 1 do
        if d[i] == nil then
            Echo("%d nil", i)                       -- x nil
        else
            Echo("%d %s", d[i].index, d[i].name)    -- x Display x
        end
    end
```

From oPC with 2 displays I see both of the displays, but running the same plugin on the web remote shows 1 (as expected) but 2 if calling via the macro.

Feels like a bug, but it could just be how it works, I'm unsure. perhaps someone on the official forums.

**Comment by lukas-runge** [comment, marked as answer]
Hey @hossimo, hey everyone,

it seems like not being able to call plugins from a layout view is indeed a missing feature. At least for the gui part of things. Thanks to the help of **Andreas Glad** I found a way of manipulating the default action of plugins assigned to Layout View to be `Call`. This enables calling the plugin from within the right context and thus renders the `PopupInput` on the webremote 🎉 

`Set Root ShowData.UserProfiles.*.LayoutElementDefaults*.3 Property Action "Call"`

After that the plugin is properly called. 😍
Huge thanks to **Andreas Glad** and **@hossimo** for your help on this issue!

Kind regards
@lukas-runge
